game/theater/theatergroundobject.py

```diff
@@ -481,18 +488,64 @@ def __init__(
             dcs_identifier="AA",
             sea_object=False,
         )
-        # Set by the SAM unit generator if the generated group is compatible
-        # with Skynet.
-        self.skynet_capable = False
+        self.aa_ranges: List[AirDefenseRange] = [aa_range]
 
     @property
+    # Returns the TGO Name
     def group_name(self) -> str:
-        if self.skynet_capable:
-            # Prefix the group names of SAM sites with the side color so Skynet
-            # can find them.
-            return f"{self.faction_color}|SAM|{self.group_id}"
+        return super().group_name
+
+    # Returns the name of one single group
+    def groups_name(self, group: Group) -> str:
+        try:
+            group_index = self.groups.index(group)
+            if self.skynet_capability_for_range(
+                self.aa_ranges[group_index], (group_index > 0)
+            ):
+                return self.skynet_group_name(group_index)
+        except AttributeError:  # Allow backwards compatibility
+            return self.skynet_group_name()
+        except ValueError:  # Allow backwards compatibility
+            pass
+
+        return super().group_name
```

Comment:
Won't this prevent the mission from generating? In the fallback case won't each group get the same name?

Reply:
really good catch! thanks